sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -242,8 +242,11 @@ public T To<T>(JsonSerializerOptions options)
             return JsonSerializer.Deserialize<T>(ToString(), options);
         }
 
-        /// <inheritdoc />
-        public override string ToString()
+        /// <summary>
+        /// Returns a stringified version of the JSON for this value.
+        /// </summary>
+        /// <returns>Returns a stringified version of the JSON for this value.</returns>
+        public string ToJsonString()
```

Comment:
Why the split?

Reply:
The behavior of JSON serializing the in .ToString() makes it unpleasant to use in the common case.  For example, if you are pull some string property out of a document, something like:

```csharp
Console.WriteLine($"The value of property 'foo' is {e["foo"]}");
```

If the default .ToString JSON serializes, you'll print `The value of property 'foo' is "bar"`, with an extra set of quotes.  This felt bad and a place where it would end up surprising folks.

Reply:
But this type is explicitly to support JSON values. After this split ToString would work correctly for some JSON i.e. `{}` or '[]' but not others and I'm not sure customers would understand the difference.

On the other hand, it seems to be what JsonElement does as well: https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.tostring?view=net-5.0#System_Text_Json_JsonElement_ToString

Reply:
Keeping it this way for now, but will think about this some more (and look into what the BCL is doing for their mutable DOM type, which we can hopefully adopt)